kitchen/src/main/java/com/zeecoder/kitchen/service/KitchenService.java

```diff
@@ -21,37 +18,41 @@
 @Service
 @RequiredArgsConstructor
 public class KitchenService {
-
-    private final Producer<WorkerState> event;
-    private final Producer<DeliveryEvent> producer;
+    private final Producer<OrderEvent> orderEventProducer;
+    private final Producer<StateEvent> stateEventProducer;
+    private final Producer<StatisticEvent> statisticEventProducer;
```

Comment:
What do you think about moving logic of producers creation to separate class like factory?

Reply:
I can use factory design pattern here?

Reply:
Yep, why not? You will inject 1 parameter instead of 3 with one function that will return implementation of Producer<T> from DI depending on T